### PR TITLE
boolean: initial commit

### PR DESCRIPTION
**Pull request by tom-wa**
# Purpose

canonicalizes boolean values, see #308

### CODE REVIEW

File: src/plugins/boolean/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+- infos = Information about the boolean plugin is in keys below
+- infos/author = Name <name@libelektra.org>
```
Comment (markus2330) [inline src/plugins/boolean/README.md]: Author missing

File: src/plugins/boolean/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+- infos/licence = BSD
+- infos/needs =
+- infos/provides =
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = recommended productive maintained reviewed conformant compatible coverage specific unittest shelltest tested nodep libc configurable preview nodoc concept
```
Comment (markus2330) [inline src/plugins/boolean/README.md]: Explanations+status missing

File: src/plugins/boolean/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+check/type = boolean
+[k6]
+check/type = boolean
+[k7]
+check/type = boolean
+[k8]
+check/type = boolean
+
+% kdb umount /booltest
+% kdb mount bool.dump /booltest dump boolean false='FALSE; OFF; 0',true='TRUE; ON; 1',set/true='this should be true',set/false='this should be false'
```
Comment (markus2330) [inline src/plugins/boolean/README.md]: How does it work with default config?

File: src/plugins/boolean/boolean.c [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+		++ptr;
+	}
+	free (localString);
+	return retVal;
+}
+
+static void normalize (Key * key, BoolData * data)
+{
+	const char * defaultTrueValues[] = {
+		"TRUE", "1", "ON", NULL,
```
Comment (markus2330) [inline src/plugins/boolean/boolean.c]: add more variants? e.g. ENABLE(D), YES

File: src/plugins/boolean/boolean.c [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+	};
+	const char * defaultFalseValues[] = {
+		"FALSE", "0", "OFF", NULL,
+	};
+	const char * value = keyString (key);
+	const char ** falseValues = (const char **)data->falseValues;
+	const char ** trueValues = (const char **)data->trueValues;
```
Comment (markus2330) [inline src/plugins/boolean/boolean.c]: e.g. DISABLE(D), NO(T)

File: src/plugins/boolean/boolean.c [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+	return retVal;
+}
+
+static void normalize (Key * key, BoolData * data)
+{
+	const char * defaultTrueValues[] = {
+		"TRUE", "1", "ON", NULL,
+	};
+	const char * defaultFalseValues[] = {
+		"FALSE", "0", "OFF", NULL,
```
Comment (markus2330) [inline src/plugins/boolean/boolean.c]: e.g. DISABLE(D), NO(T)

File: src/plugins/boolean/boolean.c [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+		KeySet * config = elektraPluginGetConfig (handle);
+		data = elektraCalloc (sizeof (BoolData));
+		parseConfig (config, data);
+		elektraPluginSetData (handle, data);
+	}
+	Key * key;
+	ksRewind (returned);
+	while ((key = ksNext (returned)) != NULL)
+	{
+		const Key * boolMeta = keyGetMeta (key, "check/type");
```
Comment (markus2330) [inline src/plugins/boolean/boolean.c]: can you also use "type" (used for json)

File: src/plugins/boolean/boolean.c [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+	if (isTrue (value, trueStrings))
+	{
+		keySetMeta (key, "origvalue", keyString (key));
+		keySetString (key, data->true);
+	}
+	else if (isFalse (value, falseStrings))
+	{
+		keySetMeta (key, "origvalue", keyString (key));
+		keySetString (key, data->false);
+	}
```
Comment (markus2330) [inline src/plugins/boolean/boolean.c]: Please add at least a warning if its neither true nor false.

It was suggested that true should be used in this case.

File: src/plugins/boolean/README.md (hunk 3 above)
Comment (tom-wa) [inline src/plugins/boolean/README.md]: default settings are 1 for true and 0 for false, and the accepted values are the same as in this example: "TRUE, ON, 1, FALSE, OFF, 0".

File: src/plugins/boolean/testmod_boolean.c [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+#include <kdbconfig.h>
+
+#include <tests_plugin.h>
+
+
+int main (int argc, char ** argv)
+{
+	printf ("BOOLEAN     TESTS\n");
+	printf ("==================\n\n");
```
Comment (markus2330) [inline src/plugins/boolean/testmod_boolean.c]: Please add tests.

File: src/plugins/boolean/README.md (hunk 3 above)
Comment (markus2330) [inline src/plugins/boolean/README.md]: I mean that is what should be explained in the docu.

**Pull request merged**